tests/test_routes.py

```diff
@@ -522,6 +534,112 @@ def test_delete_address_invalid_request(self):
+
+        # update the Customer
+        dummy_email =  "User759@gmail.com";
+        updated_customer = response.get_json()
+        logging.debug(updated_customer)
+        updated_customer["email"] = dummy_email
+        response = self.client.put(f"{BASE_URL}/{updated_customer['id']}", json=updated_customer, content_type="custom_content_type_v100")
+        self.assertEqual(response.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+
+    ##########################################
+    # UPDATE ADDRESS
+    ##########################################
+    
+    def test_update_customer_address(self):
+        """It should Update an existing Customer Address"""
+        
+        # create a Customer with 3 addresses to update
+        customer = CustomerFactory()
+        addresses = AddressFactory.create_batch(3)
+        customer.addresses.extend(addresses)
+        customer.create()
+        customer_id = customer.id
+        logging.debug(customer)
+        # Check that customer been created
+        customer = Customer.find(customer_id)
+        self.assertIsNotNone(customer)
+        # check that addresses been created
+        self.assertEqual(len(customer.addresses), 3)
+        
+        # Pick an address randomly
+        address_id_to_update = random.choice(customer.addresses).address_id
+        # update the address
+        address = Address.find(address_id_to_update)
+        self.assertIsNotNone(address)
+        dummy_street =  "Washington Square";
+        logging.debug(address)
+        address = address.serialize()
+        address["street"] = dummy_street
+        
+        response = self.client.put(f"{BASE_URL}/{address['customer_id']}/addresses/{address['address_id']}", json=address)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        updated_address = Address.find(address['address_id'])
+        logging.debug(updated_address)
+        self.assertEqual(updated_address.street, dummy_street)
+    
+    def test_update_customer_address_invalid_customer(self):
+        """It should not update address for an invalid Customer"""
+        # create a Customer with an address to update
+        customer = CustomerFactory()
+        address = AddressFactory()
+        customer.addresses.append(address)
+
+        response = self.client.put(f"{BASE_URL}/{customer.id}/addresses/{address.address_id}", json=address.serialize())
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_customer_address_invalid_address(self):
+        """It should not update invalid address for a Customer"""
+        # create a Customer with an address to update
+        customer = CustomerFactory()
+        customer.create()
+        # check customer being created
+        customer = Customer.find(customer.id)
+        self.assertIsNotNone(customer)
+        self.assertEqual(len(customer.addresses),0)
+
+        address = AddressFactory()
```

Comment:
Can omit this if we're using address_id 1 in the API call OR can use address.address_id in the next line.